docs/guides/moodleapp/development/plugins-development-guide.md

```diff
@@ -0,0 +1,2009 @@
+
+Congratulations, you have written your first Moodle plugin with moodle support!
+
+You can read the rest of this page to learn more about mobile plugins and start working on your plugin. Here's some things to keep in mind:
+
+- If you change the `mobile.php` file, you will have to refresh the browser. And remember to [disable the network cache](https://developer.chrome.com/docs/devtools/network/reference/#disable-cache).
+- If you change an existing template or function, you won’t have to refresh the browser. In most cases, doing a PTR (Pull To Refresh) in the page that displays the template will suffice.
+- If any of these doesn't show your changes, you may need to [purge all caches](https://docs.moodle.org/311/en/Developer_tools#Purge_all_caches) to avoid problems with the auto-loading cache.
+- Ultimately, if that doesn't work either, you may have to log out from the site and log in again. If any changes affect plugin installation, you may also need to increase the version in your plugin's `version.php` file and upgrade it in the site.
+
+## Types of plugins
+
+There are 3 types of plugins:
+
+### Templates generated and downloaded when the user opens the plugins
+
+With this type of plugin, the template of your plugin will be generated and downloaded when the user opens the plugin in the app. This means that your function will receive some context parameters. For example, if you're developing a course module plugin you will receive the `courseid` and the `cmid` (course module ID). You can see the list of delegates that support this type of plugin in the [Delegates](#delegates) section.
+
+![](../_files/Templates_downloaded_when_requested.png)
+
+### Templates downloaded on login and rendered using JS data
+
+With this type of plugin, the template for your plugin will be downloaded when the user logs in into the app and will be stored in the device. This means that your function will not receive any context parameters, and you need to return a generic template that will be built with JS data like the ones in the Moodle App. When the user opens a page that includes your plugin, your template will receive the required JS data and your template will be rendered. You can see the list of delegates that support this type of plugin in the [Delegates](#delegates) section.
+
+![](../_files/Templates_downloaded_on_login.png)
+
+### Pure JavaScript plugins
+
+You can always implement the whole plugin yourself using JavaScript instead of using our API. In fact, this is required if you want to implement some features like capturing links in the Mobile app. You can see the list of delegates that only support this type of plugin in the [section.
+
+## Step by step example
+
+In this example, we are going to update an existing plugin, the [Certificate activity module](https://github.com/mdjnelson/moodle-mod_certificate), that previously used a [[Moodle_Mobile_2_(Ionic_1)_Remote_add-ons|Remote add-on]] (a legacy approach to implement mobile plugins).
```

Comment:
Question: Are you planning to migrate [[Moodle_Mobile_2_(Ionic_1)_Remote_add-on]] or will this page always point to a page in the old devdocs that won't be migrated? I'm asking that because, if you are not planning this page, you should add the {{Template:WillNotMigrate}} in the MediaWiki page.

Reply:
I'll do that! Thanks!!